target/scripts/check-for-changes.sh

```diff
@@ -50,195 +53,231 @@ do
 
   # get chksum and check it, no need to lock config yet
   _monitored_files_checksums >"${CHKSUM_FILE}.new"
-  cmp --silent -- "${CHKSUM_FILE}" "${CHKSUM_FILE}.new"
-  # cmp return codes
-  # 0 – files are identical
-  # 1 – files differ
-  # 2 – inaccessible or missing argument
-  if [ $? -eq 1 ]
+  CMP_RESULT=$(cmp -- "${CHKSUM_FILE}" "${CHKSUM_FILE}.new")
+  if [[ -n "${CMP_RESULT}" ]] # If there is a difference between checksum files
```

Comment:
I know it was there before, but do some know what the `--` are for? I couldn't find something in the man page.

Reply:
> A double-dash in a shell command signals the end of options and disables further option processing.

Reply:
Commonly used for example when you're trying to `grep` something that starts with a double dash, you'd write

``` BASH
grep -E -- "--to-grep" $FILE
```

Reply:
I knew that. I was wondering, if in that case, it has another purpose. Because it seems to me unnecessary here.

Reply:
I never use it personally :)